### PR TITLE
[#9252] Javadoc: Fix Inspection Results report by IntelliJ

### LINKED ISSUE #9252: Javadoc: Fix Inspection Results report by IntelliJ
image

image

### PR DESCRIPTION
**Pull request by dh258**
Fixes #9252

Remove Javadoc error and warnings by changing some @links to @see, added @return description on parts of code, and changing some javadoc headers to block comments.

### CODE REVIEW

File: src/main/java/teammates/ui/controller/LoginServlet.java [hunk 1]
```diff
@@ -11,8 +11,8 @@
 import teammates.logic.api.GateKeeper;
 
 @SuppressWarnings("serial")
-/**
```
Comment (xpdavid) [inline src/main/java/teammates/ui/controller/LoginServlet.java]: This is actually a valid javadoc. It should be above the `@SuppressWarnings`.

File: src/main/java/teammates/ui/pagedata/PageData.java [hunk 2]
```diff
@@ -990,7 +990,7 @@ protected boolean isResponseVisibleTo(FeedbackParticipantType participantType, F
      * @param commentsForResponses map where key is response id and value is list of comments on that response
      * @param responseId response id of response associated with comment
      * @param isEditDeleteEnabled true if comment can be edited or deleted
-     * @return
+     * @return feedback response comment with associated parameters
```
Comment (xpdavid) [inline src/main/java/teammates/ui/pagedata/PageData.java]: It is not `feedback response comment with associated parameters`. The `FeedbackResponseCommentRow` is a helper class for UI display. You can just remove the `@return`.

File: src/test/resources/test.template.properties [hunk 3]
```diff
@@ -1,74 +0,0 @@
-#-----------------------------------------------------------------------------
```
Comment (xpdavid) [inline src/test/resources/test.template.properties]: Why is this get deleted?

Comment (dh258) [reply src/test/resources/test.template.properties]: Deleted by accident. Sorry

**Comment by dh258** [comment]
Ready for review

File: src/main/java/teammates/logic/api/GateKeeper.java [hunk 4]
```diff
@@ -255,7 +248,7 @@ public void verifyAccessible(InstructorAttributes instructor, FeedbackSessionAtt
      * privilegeName for feedbackSession.
      */
     public void verifyAccessible(InstructorAttributes instructor, FeedbackSessionAttributes feedbacksession,
-                                 boolean creatorOnly, String privilegeName) {
```
Comment (xpdavid) [inline src/main/java/teammates/logic/api/GateKeeper.java]: You should create your branch from the `master`. There should not be other's commit here.

Comment (dh258) [reply src/main/java/teammates/logic/api/GateKeeper.java]: Forgot to merge. Sorry

**Comment by dh258** [comment]
Ready for review

File: src/main/java/teammates/logic/api/GateKeeper.java (hunk 4 above)
Comment (xpdavid) [reply src/main/java/teammates/logic/api/GateKeeper.java]: The change still exist?

**Comment by xpdavid** [comment]
Close due to inactivity. @dh258 Free feel to re-open it.

**Pull request closed without merging**